packages/rpcs/wallet_sendTransaction/index.js

```diff
@@ -0,0 +1,230 @@
+import {getTxHashFromRawTx} from '@fluent-wallet/signature'
+import {ERROR} from '@fluent-wallet/json-rpc-error'
+import {CFX_MAINNET_NAME} from '@fluent-wallet/consts'
+import {BigNumber} from '@ethersproject/bignumber'
+
+export const NAME = 'wallet_sendTransaction'
+
+export const schemas = {
+  input: [or, cfxSchema.input, ethSchema.input],
+}
+
+export const permissions = {
+  external: ['popup', 'inpage'],
+  methods: [
+    'cfx_signTransaction',
+    'eth_signTransaction',
+    'eth_blockNumber',
+    'cfx_gasPrice',
+    'wallet_addPendingUserAuthRequest',
+    'wallet_userApprovedAuthRequest',
+    'wallet_userRejectedAuthRequest',
+    'wallet_handleUnfinishedCFXTx',
+    'wallet_handleUnfinishedETHTx',
+    'wallet_enrichConfluxTx',
+    'wallet_enrichEthereumTx',
+  ],
+  db: ['findAddress', 'getAuthReqById', 'getAddrTxByHash', 't'],
+}
+
+export const main = async ({
+  Err: {InvalidParams, Server},
+  db: {findAddress, getAuthReqById, getAddrTxByHash, t},
+  rpcs: {
+    wallet_userRejectedAuthRequest,
+    wallet_enrichConfluxTx,
+    wallet_enrichEthereumTx,
+    eth_blockNumber,
+    cfx_gasPrice,
+    cfx_signTransaction,
+    eth_signTransaction,
+    wallet_addPendingUserAuthRequest,
+    wallet_userApprovedAuthRequest,
+    wallet_handleUnfinishedCFXTx,
+    wallet_handleUnfinishedETHTx,
+  },
+  params,
+  _inpage,
+  app,
+  network,
+}) => {
+  let handleUnfinishedTxFn, signTxFn, enrichTxFn
+  if (network.type === 'cfx') {
+    handleUnfinishedTxFn = wallet_handleUnfinishedCFXTx
+    signTxFn = cfx_signTransaction
+    enrichTxFn = wallet_enrichConfluxTx
+  } else {
+    handleUnfinishedTxFn = wallet_handleUnfinishedETHTx
+    signTxFn = eth_signTransaction
+    enrichTxFn = wallet_enrichEthereumTx
+  }
+
+  if (_inpage) {
+    if (params.authReqId) throw InvalidParams('Invalid tx data')
+
+    const [{from}] = params
+
+    // check that from address is authed to the app
+    if (
+      !findAddress({
+        value: from,
+        appId: app.eid,
+      })?.length
+    )
+      throw InvalidParams(`Invalid from address in tx ${from}`)
+
+    delete params[0].nonce
+    try {
+      // try sign tx
+      await signTxFn({errorFallThrough: true}, [...params, {dryRun: true}])
+    } catch (err) {
+      if (err?.code === ERROR.USER_REJECTED.code) throw err
+      err.message = `Error while processing tx.\nparams:\n${JSON.stringify(
+        params,
+        null,
+        2,
+      )}\nerror:\n${err.message}`
+
+      throw err
+    }
+
+    if (
+      network.type === 'cfx' &&
+      app.currentNetwork.name === CFX_MAINNET_NAME
+    ) {
+      try {
+        const gasPrice = await cfx_gasPrice({errorFallThrough: true}, [])
+        if (
+          BigNumber.from(gasPrice).gt(
+            BigNumber.from(params[0].gasPrice || '0x0'),
+          )
+        ) {
+          params[0].gasPrice = gasPrice
+        }
+      } catch (err) {} // eslint-disable-line no-empty
```

Comment:
I think we should handle the error or print it or add it to sentry, otherwise, we will not know the error information if the error happens

Reply:
I think it's a product issue.

Reply:
Unexpected errors are handled at https://github.com/Conflux-Chain/helios/blob/d22c3e6831f7f60024fca199149aad168613b7cb/packages/rpc-engine/src/error.js#L22

these are expected error